source/material_model/visco_plastic.cc

```diff
@@ -263,7 +263,13 @@ namespace aspect
               }
             }
 
-          // Step 1d: multiply the viscosity by a constant (default value is 1)
+          // Step 1d-1: compute viscosity from Peierls creep law
+          const double viscosity_peierls = peierls_creep.compute_viscosity(edot_ii, pressure, temperature_for_viscosity, j);
+
+          // Step 1d-2: harmonically average diffusion/dislocation/composite and peierls creep viscosity
+          viscosity_pre_yield = (viscosity_pre_yield * viscosity_peierls) / (viscosity_pre_yield + viscosity_peierls);
+
```

Comment:
This should be behind a `if (use_peierls_creep)`. Also like I asked above: Is this actually correct for the composite viscosity? Shouldnt be Peierls creep added in line 255? Or is this mathematically equivalent?

Reply:
Doh, that was the whole point of adding that parameter.